### ISSUE TITLE
Add Sinon.assert.calledOnceWithExactly 

### ISSUE THREAD

**Issue by dpmott** [sinonjs/sinon#2006]
**Is your feature request related to a problem? Please describe.**
Closely related to https://github.com/sinonjs/sinon/issues/1247, but in this case I'd like to be able to type `Sinon.assert.calledOnceWithExactly(stub, arg1, ... argN)`

**Describe the solution you'd like**
I believe that there just needs to be an entry in `assert.js` to facilitate this.  I'm not entirely sure what the failure message (including interpolated variables) should be though.

Possibly just this addition to assert.js?
`mirrorPropAsAssertion("calledOnceWithExactly", "expected %n to be called once with exact arguments %D");
`

**Describe alternatives you've considered**
The equivalent functionality with today's Sinon is:
```
Sinon.assert.calledOnce(stub);
Sinon.assert.calledWithExactly(stub, arg1, ...argN);
```

If I could get some agreement on what the assertion message should be, then I'd be happy to generate a PR.

**Comment by brandonau24** [comment]
@dpmott This is possible with a [spy](https://sinonjs.org/releases/latest/spies/) assuming you're on the latest version. The beginning of the page says, spies can record arguments which is what you seem to be looking for. [Stubs](https://sinonjs.org/releases/latest/stubs/) are only for pre-programming behavior like forcing a function to return a specific value. Hope this helps!

**Comment by fatso83** [comment]
@brandonau24 I believe you are talking about something else. @dpmott is addressing the feature disparity between the assertions available on the spies and the standalone assertions that can be invoked on spies. Furthermore, everything a spy can do, a stub can do as well: a stub simply extends the spy api. So all the spy assertions are already available on a stub, so you can always use a stub in place of a spy - it's more of a semantical discussion whether or not to explicitly use a spy.

@dpmott I'd try to make the assertion message even clearer (ref 88a7150 and the whole discussion in https://github.com/sinonjs/sinon/issues/1971#issuecomment-496150383).

```diff
+ What about "expected %n to be called once and with the exact arguments %D"?
- What about "expected %n to be called once with exact arguments %D"?
```

**Comment by dpmott** [comment]
Thanks for the input. Will do.

**Comment by fatso83** [comment]
Closed by #2080 !
